libs/shared/utils/http/src/lib/models/pagination/paging-request.model.ts

```diff
@@ -87,6 +87,15 @@ export class PagingRequestModel implements SchemaModel<IPagingRequest, string> {
     });
   }
 
+  public resetToStart() {
+    this.pageNumber = 0;
+    this._router?.navigate([], {
+      queryParams: { pageNumber: this.pageNumber, pageSize: this.pageSize, sortBy: this._sortBy, sortOrder: this.sortOrder } as Params,
+      queryParamsHandling: 'merge',
+      relativeTo: this._activatedRoute,
+    });
+  }
```

Comment:
A couple of things here:

- This navigation is being duplicated a bunch here we should to this in a private _function
- We should make this more reusable provide optional overrides in the reset method

```suggestion
  public reset(paging?: {
    pageNumber: 0,
    pageSize: 50,
    sortBy: '',
     sortOrder: 'ASC'
  }) {
    this.pageNumber = pageSize;
    this._router?.navigate([], {
      queryParams: { pageNumber: this.pageNumber, pageSize: this.pageSize, sortBy: this._sortBy, sortOrder: this.sortOrder } as Params,
      queryParamsHandling: 'merge',
      relativeTo: this._activatedRoute,
    });
  }
```